src/Windowing/Silk.NET.Windowing.Desktop/GlfwWindow.cs

```diff
@@ -0,0 +1,612 @@
+// This file is part of Silk.NET.
+// 
+// You may modify and distribute Silk.NET under the terms
+// of the MIT license. See the LICENSE file for details.
+
+using System;
+using System.Drawing;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Silk.NET.GLFW;
+using Silk.NET.Windowing.Common;
+using Ultz.Dispatcher;
+
+namespace Silk.NET.Windowing.Desktop
+{
+    /// <summary>
+    /// A Silk.NET window, using GLFW as a backend.
+    /// </summary>
+    public class GlfwWindow : IWindow
+    {
+        // Glfw stuff
+        private Glfw glfw = GlfwProvider.GLFW.Value;
+        private Dispatcher glfwThread = GlfwProvider.ThreadDispatcher;
+        private unsafe WindowHandle* WindowPtr;
+        
+        // Main loop-related things
+        
+        // Dispatcher for the update and render threads
+        private Dispatcher UpdateDispatcher;
+        private Dispatcher RenderDispatcher;
+
+        // The period for rendering and updating. In essence, the delay between events being fired off.
+        private double renderPeriod;
+        private double updatePeriod;
+
+        // The timestamp of the last update/render being run. Used to calculate delta.
+        private double renderLastTime;
+        private double updateLastTime;
+
+        // The number of frames that the window has been running slowly for.
+        private int _isRunningSlowlyTries;
+        
+        /// <inheritdoc />
+        public bool IsRunningSlowly => _isRunningSlowlyTries > 5;
+
+        private bool _isVisible;
```

Comment:
Are these fields ever updated? If not, we should either call GLFW in all of the getters as long as the performance impact isn't too bad. And even if it is, the user should always have the most up-to-date values. The performance impact will only affect the user where they need it, so I'm happy with telling end users to cache it themselves.

Reply:
All the cache fields are updated properly, as far as I know.